source/ui_main_window.py

```diff
@@ -61,45 +63,70 @@ def init_ui(self: QMainWindow) -> None:
 
         # print(action_nullclines.isChecked())
 
+        # Canvas to show the phase plot as part of the main window
+        # By default, open application displaying a two dimensional system
+        self.default_dims = 2
+        self.psp_canvas_default(self.default_dims)
+
         # Window Features
-        self.x_prime_label = QLabel("x' =")
-        self.y_prime_label = QLabel("y' =")
-        self.x_prime_entry = QLineEdit("y")
-        self.y_prime_entry = QLineEdit("-x")
+        self.x_prime_label = QLabel(self.phase_plot.system.system_coords[0] + "' =")
+        self.y_prime_label = QLabel(self.phase_plot.system.system_coords[1] + "' =")
+        self.x_prime_entry = QLineEdit(self.phase_plot.system.ode_expr_strings[0])
+        self.y_prime_entry = QLineEdit(self.phase_plot.system.ode_expr_strings[1])
         self.plot_button = QPushButton("Plot")
 
-        # Canvas to show the phase plot as part of the main window
-        self.phase_plot = DefaultCanvas()
-        self.phase_plot.update_system(self.phase_plot.default_system)
-
         # Nullclines are set to toggle with the "Plot Nullclines" menu option
         self.action_nullclines.changed.connect(self.phase_plot.toggle_nullclines)
 
         # Parameter inputs
+
+        param_names = list(self.setup_dict["params"].keys())
+        param_vals = list(self.setup_dict["params"].values())
+
         self.parameter_input_boxes = {}
         self.no_of_params = 5  # Number of user defined parameters
         for param_num in range(self.no_of_params):
-            self.parameter_input_boxes[
-                "param_" + str(param_num) + "_name"
-            ] = QLineEdit()
-            self.parameter_input_boxes["param_" + str(param_num) + "_val"] = QLineEdit()
+            # Fills parameter input boxes with parameter vars and corresponding vals
+            if param_num < len(self.setup_dict["params"].keys()):
+                self.parameter_input_boxes[
+                    "param_" + str(param_num) + "_name"
+                ] = QLineEdit(param_names[param_num])
+                self.parameter_input_boxes[
+                    "param_" + str(param_num) + "_val"
+                ] = QLineEdit(str(param_vals[param_num]))
+            # Allows for the situation where self.no_of_params > len(self.setup_dict["params"].keys())
+            else:
```

Comment:
What happens in the case where there are more params than param boxes? Should probably throw an error in this case.

Reply:
In the case where there are more parameters in the system than param boxes, the plot button will be rendered ineffective and won't do anything. The specific error handling hasn't been implemented in this branch yet as it would require a significant amount of work; enough to justify its own branch IMO.